### PR TITLE
add string conversion for EXP

### PR DESCRIPTION
**Pull request by WalterBright**
Splitting it off from the `Token.tochars` table.

### CODE REVIEW

File: src/dmd/hdrgen.d [hunk 1]
```diff
@@ -3962,3 +3962,152 @@ private void typeToBufferx(Type t, OutBuffer* buf, HdrGenState* hgs)
...
     }
 }
+
+/****************************************
+ * Convert EXP to char*.
+ */
+
+string EXPtoString(EXP op)
+{
+    __gshared immutable char*[EXP.max + 1] strings =
```
Comment (RazvanN7) [inline src/dmd/hdrgen.d:3972]: dmd/src/dmd/hdrgen.d(3972:44)[warn]: Variable declaration for `strings` has redundant attributes (`__gshared`, `immutable`)

Comment (RazvanN7) [reply src/dmd/hdrgen.d:3972]: @WalterBright I see that you made another push, however, the buildkite failure persists as this was not fixed.

Comment (ljmf00) [reply src/dmd/hdrgen.d:3972]:
```suggestion
    static immutable char*[EXP.max + 1] strings =
```

**Comment by WalterBright** [comment]
semaphoreci continues it's remarkable ability to randomly fail with no clue as to what failed and where

**Comment by ljmf00** [comment]
> semaphoreci continues it's remarkable ability to randomly fail with no clue as to what failed and where

Seems unrelated.

**Review by ljmf00 — APPROVED** [review]
Looks good.

**Comment by ljmf00** [comment]
> > semaphoreci continues it's remarkable ability to randomly fail with no clue as to what failed and where
> 
> Seems unrelated.

Circle CI timed out too.

**Comment by WalterBright** [comment]
> Circle CI timed out too.

And that worked before I fixed the style complaint.

**Comment by maxhaton** [comment]
@WalterBright CircleCI rerun; now passes tests (click through to their website, top right corner there's a button to rerun) - these failures are because the CI job seems to end up basically running on a Casio calculator and times out (Should finish with about 4x margin to spare, somehow it uses up a whole hour)

**Pull request merged**